superset-frontend/src/SqlLab/components/ScheduleQueryButton.tsx

```diff
@@ -0,0 +1,208 @@
+`;
+
+const ButtonComponent = styled.div`
+  background-color: none;
+  .ant-btn.superset-button {
+    color: rgba(0, 0, 0, 0.85);
+    text-transform: none;
+    padding-right: 0px;
+    padding: 0px;
+    font-size: 14px;
+    height: 22px;
+    font-weight: ${({ theme }) => theme.typography.weights.normal};
+    &:first-child {
+      background: none;
+    }
+  }
+`;
+
+const ScheduleQueryButton: FunctionComponent<ScheduleQueryButtonProps> = ({
+  defaultLabel = t('Undefined'),
+  sql,
+  schema,
+  dbId,
+  onSchedule = () => {},
+  scheduleQueryWarning = null,
+  tooltip = '',
+}) => {
+  const [description, setDescription] = useState('');
+  const [label, setLabel] = useState(defaultLabel);
+  const [showSchedule, setShowSchedule] = useState(false);
+  let saveModal: any;
+
+  const onScheduleSubmit = ({ formData }: Record<string, any>) => {
+    const query = {
+      label,
+      description,
+      db_id: dbId,
+      schema,
+      sql,
+      extra_json: JSON.stringify({ schedule_info: formData }),
+    };
+    onSchedule(query);
+    saveModal.close();
+  };
+
+  const renderModalBody = () => (
+    <FormGroup>
+      <StyledRow>
+        <Col md={12}>
+          <FormLabel className="control-label" htmlFor="embed-height">
+            {t('Label')}
+          </FormLabel>
+          <FormControl
+            type="text"
+            placeholder={t('Label for your query')}
+            value={label}
+            onChange={(event: any) => setLabel(event.target?.value)}
+          />
+        </Col>
+      </StyledRow>
+      <StyledRow>
+        <Col md={12}>
+          <FormLabel className="control-label" htmlFor="embed-height">
+            {t('Description')}
+          </FormLabel>
+          <FormControl
+            componentClass="textarea"
+            placeholder={t('Write a description for your query')}
+            value={description}
+            onChange={(event: any) => setDescription(event.target?.value)}
+          />
+        </Col>
+      </StyledRow>
+      <Row>
+        <Col md={12}>
+          <div className="json-schema">
+            <Form
+              schema={getJSONSchema()}
+              uiSchema={getUISchema}
+              onSubmit={onScheduleSubmit}
+              validate={getValidator()}
+            />
+          </div>
+        </Col>
+      </Row>
+      {scheduleQueryWarning && (
+        <Row>
+          <Col md={12}>
+            <small>{scheduleQueryWarning}</small>
+          </Col>
+        </Row>
+      )}
+    </FormGroup>
+  );
+
+  return (
+    <span className="ScheduleQueryButton">
+      <ModalTrigger
+        ref={ref => {
+          saveModal = ref;
+        }}
+        modalTitle={t('Schedule query')}
+        modalBody={renderModalBody()}
+        triggerNode={
+          <ButtonComponent>
+            <Button
+              onClick={() => setShowSchedule(!showSchedule)}
+              buttonSize="small"
+              tooltip={tooltip}
```

Comment:
I think we lost `disabled` here.. also you can add `buttonStyle="link"` and it will get you most of the styles that you need.